exercises/Toluwalase_Agoro/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,154 @@
+    if (pair.first > maxCount) {
+      maxCount = pair.first;
+    }
+  }
+  std::vector <std::vector <int>> vectorForSort;
+  for (int i = 0; i < maxCount; i++) {
+    std::vector <int> emptyVector;
+    vectorForSort.push_back(emptyVector);
+  }
+
+  for (int i = 0; i < vectorOfPairs.size(); i++) {
+    int index = vectorOfPairs[i].first - 1;
+    vectorForSort[index].push_back(i);
+  }
+  std::vector <std::pair<int, std::string>> sortedVector;
+  for (int i = int(vectorForSort.size()) - 1; i >= 0; i--) {
+    for (auto& order : vectorForSort[i]) {
+      sortedVector.push_back(vectorOfPairs[order]);
+    }
+  }
+  vectorOfPairs = sortedVector;
+}
+
+/**returns a formatted word from input file
+  *
+  *@param stream the ifstream object to read from
+  *returns a formatted string
+  */
+std::string getWord(std::ifstream& stream) {
+  
+  std::string outputString;
+  char inputCharacter;
+
+  while (true) {
+    stream.get(inputCharacter);
+    if (inputCharacter == '-' || inputCharacter == ' ' || inputCharacter == '\n' || stream.eof())
+      break;
+    if (inputCharacter == '"' || inputCharacter == '!' || inputCharacter == '('
+      || inputCharacter == ')' || inputCharacter == ':' || inputCharacter == '\''
+      || inputCharacter == ',' || inputCharacter == '?' || inputCharacter == '.')
+      continue;
+
+    outputString += tolower(inputCharacter);
+  }
+  return outputString;
+}
+
+int main() {
+  /*This block of code declares all vectors used
+  and gets input from user so a to open
+  input file
+  */
+  
+
+  char filePathToInput[150];
+  std::cout << "Please input a file path to input file :";
+  std::cin.getline(filePathToInput, 150);
+  std::ifstream fileToRead;
+  fileToRead.open(filePathToInput);
+  std::vector <std::pair<int, std::string>> vectorOfCountsAndWords;
+
+  if (!fileToRead) {
+    /* This if block checks if file was opened correctly
+    and if not program is terminated
+    */
+    std::cout << "File could not be opened" << std::endl;
+    std::cout << "Ensure that file path is valid" << std::endl;
+  }
+  std::string holderString;
+  auto myPredictor = [&holderString](std::pair<int, std::string>& pairOfCountAndWord)
+  {
+    return (pairOfCountAndWord.second == holderString);
+  };
+
+  while (!fileToRead.eof()) {
+    /* This while block reads input file and adds formatted words to
+    VectorOfWords until the end of file has been reached. This block
+    has an if , else if block that determines whether word is more than
+    4 letters and already exists in VectorOfWords. if it does it increments
+    the count in VectorOfCounts otherwise it is added to VectorOfWords
+    */
+    
+    holderString = getWord(fileToRead);
+    auto location = std::find_if(vectorOfCountsAndWords.begin(),vectorOfCountsAndWords.end(), myPredictor);
+    if (holderString.size() > 4 && location == vectorOfCountsAndWords.end()) {
+      vectorOfCountsAndWords.push_back(std::make_pair(1,holderString));
```

Comment:
Just thought of one other thing that we're trying to encourage. We are in general trying to use the function emplace_back instead of push_back when adding items to vectors. The reason is that it has better performance when temporary objects are added as in this case.
To use emplace back to construct an object in place you can just pass the arguments to the constructor of the pair:
vectorOfCountsAndWords.emplace_back(1,holderString);
If you've not pushed your latest set of changes yet please could you make this change also?